### PR TITLE
Feature/maps user profile

### LINKED ISSUE #1958: Maps in User Profile
- [x] Show road map instead of satellite
- [x] Center Map so that more than one items are visible.
If an individual has planted in more than one continent, zoom out the map, so that all contributions are visible.

- [x]  Improve the rendering of the page. Scrolling the map on devices with low memory feels very sluggish.

- [x] Add full screen button. Please use icons, instead of png, if possible.

Screen Shot 2020-02-15 at 8 31 10 PM

- [x] Full Screen View:

Screen Shot 2020-02-15 at 8 40 00 PM

Design for cards are just below the Map.
https://zpl.io/VqX9QG6

### PR DESCRIPTION
**Pull request by haideralishah**
- [x] Back button is often unresponsive in the Maps/ Single Contribution Page, Improve the button performance.

- [x] Clicking on tree icon on map, often takes to wrong/different contribution. This has been fixed for navigation from cards, but not from icons.

- [x] Loading a single contribution fetches all projects related to contributions. instead it should fetch neighbors only When contribution cards are stacked as `A [B] C` B being the current view, only load projects related to A, B & C, (performance)

- [x] Remove Current Location Icon from the Map View

- [x] Change to a icon of a tree for map marker instead of image (performance)

- [ ] Use Apple Maps in iOS ( if it is more performant than google maps)
- [x] disable the marker press if detail not show
- [x] revert the `headerText` on outer card list
- [x] disable arrows keys on both ends 
- [x] clicking on multiple card on outer list
- [x] Indexing issues
- [x] In Single Contribution, enable Satellite View by default.

Things to consider:

- [x] Look into Map Clustering

------------
Fixes #1958

Changes in this pull request:
- Show road map instead of satellite
- Center Map so that more than one items are visible.

@sagararyal

### CODE REVIEW

File: app/components/UserHome/FullMapComponent.js [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+  View,
+  Animated,
+  Image,
+  Dimensions,
+  TouchableOpacity
+} from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import MapView from 'react-native-maps';
+import { tree_1 } from '../../assets/index';
+import Smalltreewhite from '../../assets/images/smalltreewhite.png';
```
Comment (norbertschuler) [inline app/components/UserHome/FullMapComponent.js]: Is there any special reason you don't load this asset from `assets/index` as e.g. `tree_1`?

File: app/components/UserHome/FullMapComponent.js [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+      onPress={() => toNavigateUserContributionDetail(marker)}
+      style={styles.cardContainer}
+    >
+      <View style={{ flex: 1 }}>
+        <View>
+          <Text style={styles.cardHeaderText}>Gift to Sam William</Text>
+        </View>
+        <View style={{ marginVertical: 5 }}>
+          <Text style={styles.subHeaderText}>
+            Planted by Eden Reforestation Projects
```
Comment (norbertschuler) [inline app/components/UserHome/FullMapComponent.js]: Is this still work in progress with the sample text elements not using text resources?

Comment (haideralishah) [reply app/components/UserHome/FullMapComponent.js]: I need `Heading` and `subheading` data in single tree contribution API. I already requested @harshvitra, @ankitecd, and @jmiridis to add this data in API so I will replace it with dynamic data.
no need to use text resources

File: app/components/UserHome/FullMapComponent.js (hunk 1 above)
Comment (haideralishah) [reply app/components/UserHome/FullMapComponent.js]: I used this image 
https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/397e98913da9f1cf23fa51e7705d64cd4e7d9823/app/components/UserHome/FullMapComponent.js#L248

Comment (norbertschuler) [reply app/components/UserHome/FullMapComponent.js]: I meant why do you directly import `Smalltreewhite` and not put this image into  https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/app/assets/index.js?

Comment (norbertschuler) [reply app/components/UserHome/FullMapComponent.js]: I meant why do you directly import `Smalltreewhite` and not put this image into  https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/app/assets/index.js?

Comment (haideralishah) [reply app/components/UserHome/FullMapComponent.js]: fixed

**Comment by sagararyal** [comment]
- [ ] if markers in different part of the world, zoom out to show all markers

Screen Shot 2020-03-02 at 12 38 21 PM

- [ ] Button is not circle

Screen Shot 2020-03-02 at 12 34 30 PM

- [ ] clicking full screen error

Screen Shot 2020-03-02 at 12 35 03 PM


- [ ] Card Sizes are too big
Screen Shot 2020-03-02 at 12 39 22 PM

- [ ] Remove the while space at the bottom like at top
Screen Shot 2020-03-02 at 12 40 07 PM

- [ ] Clicking a marker does not always center the map

- [ ] Use the map marker as in design

- [ ] Use Google Maps style in full screen; do not load a new layer.

- [ ] Hide button animation doesn't work as a user would expect
Screen Shot 2020-03-02 at 12 43 14 PM

- [ ] Add space above the button  for phones that use safe area view.

**Comment by haideralishah** [comment]
- Google Map Style doesn't work on Apple Maps.
- I change the Icon `down arrow` to `close`
- Combine with Single Contribution Page
currently, when users tap on any card it's navigating the single contribution detail page.

**Comment by haideralishah** [comment]
Hide button animation doesn't work as a user would expect Done.
Hnet-image

**Comment by haideralishah** [comment]
Google Map Style is now working fine on iOS.
image

**Comment by haideralishah** [comment]
Combine with Single Contribution Page
Hnet-image (1)

File: app/components/UserHome/FullMapComponent.js [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+  initiateComponent = () => {
+    this.setState(
+      {
+        markers: this.props.userContributions,
+        region: {
+          latitude: 37.78825,
+          longitude: -122.4324,
+          latitudeDelta: 0.0922,
+          longitudeDelta: 0.0421
+        }
```
Comment (norbertschuler) [inline app/components/UserHome/FullMapComponent.js]: This is somewhere in San Fransisco. @sagararyal what should be default coordinates to use in the app?

**Comment by harshvitra** [comment]
Bottom Navigation hides when user visits the Map and comes back to the Me Page

File: ios/TreecounterApp.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -30,12 +30,12 @@
 		891FA4A6378545F783E5B6A9 /* OpenSans-Light.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 4F0548F411A546C79D230135 /* OpenSans-Light.ttf */; };
 		8E91A3D298D54668A81191C0 /* OpenSans-ExtraBold.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 16506A78B0404858BF1334D1 /* OpenSans-ExtraBold.ttf */; };
 		96E3C6F17CE24BD5AFF45E7B /* OpenSans-Bold.ttf in Resources */ = {isa = PBXBuildFile; fileRef = B0362431F9D24179A7E9175E /* OpenSans-Bold.ttf */; };
-		ABCD2BA66D672482E2F5A002 /* libPods-TreecounterApp.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 34D4B68F0F12DDDBB81E2F5B /* libPods-TreecounterApp.a */; };
```
Comment (norbertschuler) [inline ios/TreecounterApp.xcodeproj/project.pbxproj]: We should just copy the file from the develop branch into this branch, so we can avoid merge back any unnecessary changes to this file. Probably it changed due to rebuilding the iOS project (or we changed it in the develop branch).

File: app/utils/getRandomImage.js [hunk 5]
```diff
@@ -11,8 +11,9 @@ import {
 } from '../assets';
 
 export default function GetRandomImage(props) {
+  console.log(props, 'propsprops');
```
Comment (norbertschuler) [inline app/utils/getRandomImage.js]: Please remove this or replace with `debug()` if you want to merge it.

**Comment by norbertschuler** [comment]
- [ ] design issue showing my trees very much at the borders - maybe zoom out a little bit:

Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 24 48

**Comment by norbertschuler** [comment]
Issues in user contribution detail screen:

- [ ] Why is there a line break though it seems there is enough space for the name?
- [ ] Nothing happens when I click the project name 

Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 27 47

- [ ] When you use the arrow to change the map to fullscreen, the name of the user contribution changes to the first user contribution in my list instead of the currently opened contribution.
- [ ] The name of the user contribution can be clicked, but nothing happens.

Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 39 25

- [ ] Switching between full map and partial map sometimes can result into this showing a single user contribution page with the map showing all user contributions:

Simulator Screen Shot - iPhone 11 - 2020-03-19 at 20 32 09

File: app/components/UserContributions/ContributionCard.native.js [hunk 6]
```diff
@@ -301,10 +300,11 @@ class ContributionCard extends React.Component {
           <TouchableOpacity
             style={styles.singleRedeemObject}
             onPress={() => {
-              this.props.navigation.navigate('contribution_details', {
-                contribution,
-                titleParam: plantProjectName || tpoName || treeSpecies
-              });
+              this.props.onPressSingleContribution(contribution.id);
```
Comment (norbertschuler) [inline app/components/UserContributions/ContributionCard.native.js]: You have not defined this new property in `ContributionCard.propTypes`, have you?

File: app/components/UserContributions/ContributionCardList.native.js [hunk 7]
```diff
@@ -1,36 +1,43 @@
...
-            )
-        );
+  return contributions.length > 3 && showAllContributions ? (
+    <FlatList
+      data={contributions}
+      renderItem={({ item }) => (
+        <ContributionCard
+          onPressSingleContribution={onPressSingleContribution}
+          contribution={item}
+          deleteContribution={deleteContribution}
```
Comment (norbertschuler) [inline app/components/UserContributions/ContributionCardList.native.js]: I can not find this functionality any more in ContributionCard? How can we delete contributions like self registered trees. Maybe be lost this somehow and not in this patch, but here I recognized it first.

File: app/components/UserHome/AnimatedMap.js [hunk 8]
```diff
@@ -0,0 +1,862 @@
...
+import { deleteContribution } from '../../actions/EditMyTree';
+import { delimitNumbers } from '../../utils/utils';
+import { loadProject } from '../../actions/loadTposAction';
+import { currentUserProfileIdSelector } from '../../selectors/index';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { getAllPlantProjectsSelector } from '../../selectors';
+import { markerImage } from '../../assets/index.js';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import { tree_1 } from '../../assets/index';
```
Comment (norbertschuler) [inline app/components/UserHome/AnimatedMap.js]: I think you can import both image assets `markerImage` and `tree_1` with one import statement.

File: app/components/UserHome/AnimatedMap.js [hunk 9]
```diff
@@ -0,0 +1,862 @@
...
+          latitudeDelta: 0.00095,
+          longitudeDelta: 0.0095
+        }
+      },
+      () => {
+        setTimeout(() => {
+          try {
+            this.mapView.fitToSuppliedMarkers(markers.map(x => String(x.id)));
+          } catch (e) {
+            console.log(JSON.stringify(e))
```
Comment (norbertschuler) [inline app/components/UserHome/AnimatedMap.js]: Please replace all console.log with `debug()` or remove them.

File: app/components/UserHome/panController.js [hunk 10]
```diff
@@ -0,0 +1,440 @@
+import React from 'react';
```
Comment (norbertschuler) [inline app/components/UserHome/panController.js]: Why did you copy this code (without mentioning the initial author in a comment) into our project? Can't we use the npm module `react-native-pan-controller` directly?

Comment (haideralishah) [reply app/components/UserHome/panController.js]: No, this is not a package this is actually an example in `react-native-maps` repo. https://github.com/react-native-community/react-native-maps/blob/master/example/examples/PanController.js

I think we can import this component like`react-native-maps/path/to/panController`
but I customise this component.

File: app/components/UserHome/AnimatedMap.js (hunk 9 above)
Comment (haideralishah) [reply app/components/UserHome/AnimatedMap.js]: removed

File: app/components/UserContributions/ContributionCardList.native.js (hunk 7 above)
Comment (haideralishah) [reply app/components/UserContributions/ContributionCardList.native.js]: not sure about it, I just reuse this component also present in develop.

File: app/components/UserHome/AnimatedMap.js (hunk 8 above)
Comment (haideralishah) [reply app/components/UserHome/AnimatedMap.js]: fixed

**Comment by haideralishah** [comment]
@ankitecd please check the following issues in this pr.
- [ ] single contribution details not loading in detail page.
- [ ] check line break & `onPress` is not working on project name.
image

**Comment by norbertschuler** [comment]
@haideralishah could you fix the eslint errors from `npm run lint:errors`:

```
/home/circleci/ttc-app/app/components/UserContributions/ContributionCard.native.js
  255:39  error  'giveeSlug' is defined but never used               no-unused-vars
  282:17  error  Strings must use singlequote                        quotes
  283:9   error  'treeCountLine' is assigned a value but never used  no-unused-vars

/home/circleci/ttc-app/app/components/UserHome/AnimatedMap.js
   10:3   error  'Text' is defined but never used                      no-unused-vars
  105:9   error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  141:17  error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  254:7   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  256:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  261:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  313:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  318:9   error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle
  325:7   error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  325:24  error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  333:7   error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  333:24  error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  334:78  error  Strings must use singlequote                          quotes
  367:7   error  'lastActiveIndex' is assigned a value but never used  no-unused-vars
  370:9   error  'activeMarker' is assigned a value but never used     no-unused-vars
  410:12  error  Unexpected constant condition                         no-constant-condition
  411:13  error  Unexpected constant condition                         no-constant-condition
  423:33  error  Unexpected dangling '_' in '_carousel'                no-underscore-dangle
  482:10  error  Unexpected constant condition                         no-constant-condition
  493:29  error  Unexpected dangling '_' in '_carouselDetail'          no-underscore-dangle

```

**Comment by norbertschuler** [comment]
- [ ] This is what it looks like for me when I maximize the map for my user contributions:

Simulator Screen Shot - iPhone 11 - 2020-04-07 at 18 12 19

**Comment by norbertschuler** [comment]
- [ ] This message on the iOS simulator occurs when I click the "get my position" button in the maximized map:
```
console.error: {"code":2,"message":"Unable to retrieve location.","PERMISSION_DENIED":1,"POSITION_UNAVAILABLE":2,"TIMEOUT":3}
```

**Comment by norbertschuler** [comment]
- [ ] Clicking on the Project name does nothing on my iPhone X simulator:

Simulator Screen Shot - iPhone 11 - 2020-04-07 at 18 20 43

File: ios/TreecounterApp/Info.plist [hunk 11]
```diff
@@ -20,6 +20,10 @@
 	<string>$(MARKETING_VERSION)</string>
 	<key>CFBundleSignature</key>
 	<string>????</string>
+	<key>NSLocationWhenInUseUsageDescription</key>
```
Comment (norbertschuler) [inline ios/TreecounterApp/Info.plist:23]: The key `NSLocationWhenInUseUsageDescription` is already part of this file, this is a duplicate.

File: ios/TreecounterApp/Info.plist [hunk 12]
```diff
@@ -20,6 +20,10 @@
...
+	<string></string>
+	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
```
Comment (norbertschuler) [inline ios/TreecounterApp/Info.plist:25]: If we need this key next to `NSLocationAlwaysAndWhenInUseUsageDescription` and `NSLocationWhenInUseUsageDescription` we need to add some text to this and also do a German translation as we haven't got this resources in our LingoHub tool, but currently translate them manually in the iOS project.

File: app/components/UserContributions/ContributionCardList.native.js (hunk 7 above)
Comment (norbertschuler) [reply app/components/UserContributions/ContributionCardList.native.js:20]: I tested again that I was able to delete my self registered trees - single and multiple. It works, but then I get directed to the home screen and when I go back to the "Me" screen, I still see the user contribution which I just deleted.

**Comment by norbertschuler** [comment]
- [ ] I am missing user contribution details as the video or the detail card as we currently show the in the `develop` branch:

Simulator Screen Shot - iPhone 11 - 2020-04-07 at 19 21 29

**Comment by sanjayradadiya** [comment]
@haideralishah when I delete register tree it's redirect me to userHome screen without any message 
https://www.loom.com/share/c445a85243a6483bafdfd539a5970b7c

**Comment by haideralishah** [comment]
> https://www.loom.com/share/c445a85243a6483bafdfd539a5970b7c

fixed

File: app/components/UserContributions/ContributionCard.native.js (hunk 6 above)
Comment (norbertschuler) [reply app/components/UserContributions/ContributionCard.native.js]: `onPressSingleContribution` still is not defined in the propTypes of `ContributionCard`, shouldn't it?

Comment (norbertschuler) [reply app/components/UserContributions/ContributionCard.native.js]: changed this myself

**Comment by norbertschuler** [comment]
- [ ] I found these two warnings in the browser console when I first open my "Me" screen:

```
Warning: Failed child context type: Invalid child context `virtualizedCell.cellKey` of type `number` supplied to `CellRenderer`, expected `string`.
    in CellRenderer (at VirtualizedList.js:767)
    in RCTScrollContentView (at ScrollView.js:1038)
    in RCTScrollView (at ScrollView.js:1178)
    in ScrollView (at VirtualizedList.js:1183)
    in VirtualizedList (at FlatList.js:676)
    in FlatList (at UserHome/index.native.js:617)
```

```
Warning: Failed prop type: The prop `data` is marked as required in `Carousel`, but its value is `null`.
    in Carousel (at AnimatedMap.js:431)
```

**Comment by norbertschuler** [comment]
- [ ] I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:

Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 14
Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 05

File: ios/TreecounterApp.xcodeproj/project.pbxproj (hunk 4 above)
Comment (norbertschuler) [reply ios/TreecounterApp.xcodeproj/project.pbxproj]: Did it myself.

File: app/utils/getRandomImage.js (hunk 5 above)
Comment (norbertschuler) [reply app/utils/getRandomImage.js]: cleanup up myself

File: ios/TreecounterApp/Info.plist (hunk 11 above)
Comment (norbertschuler) [reply ios/TreecounterApp/Info.plist]: removed it

File: ios/TreecounterApp/Info.plist (hunk 12 above)
Comment (norbertschuler) [reply ios/TreecounterApp/Info.plist]: added resource properly plus German translation

**Comment by norbertschuler** [comment]
I wonder about two further issues:

- [ ] I donated several trees in Mexico resulting into several tree icons at the same place. I see some flickering on the map where they are all located - is that a bug in the used map component?

Simulator Screen Shot - iPhone 11 - 2020-04-17 at 19 41 20

- [ ] On my overview (see above) the following trees are not shown, but I can scroll to them. Is that because we cannot zoom further out of the map to show them all?

Simulator Screen Shot - iPhone 11 - 2020-04-17 at 19 41 51

**Comment by norbertschuler** [comment]
- [ ] One more issue: When I use the button to maximize the map, I only get show the detailed map for my newest user contribution - is that's how it's wanted to be?

Simulator Screen Shot - iPhone 11 - 2020-04-17 at 19 50 00

**Comment by norbertschuler** [comment]
- [ ] Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)

**Comment by haideralishah** [comment]
> * [ ]  Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)

@norbertschuler Map on detail page should be in satellite mode 
this is a default requirement not a issue.

**Comment by norbertschuler** [comment]
> > * [ ]   Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)
> 
> @norbertschuler Map on detail page should be in satellite mode
> this is a default requirement not a issue.

Yes, but if you switch it to normal mode, why does the overview map on the user contribution list page get also switched to satellite mode? That's weird ;-)

**Comment by haideralishah** [comment]
> > > * [ ]    Another issue: Switching from satellite to map view in the user contribution details also switched the map view to satellite view for the overview map:  [map.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4493041/map.mp4.zip)
> > 
> > 
> > @norbertschuler Map on detail page should be in satellite mode
> > this is a default requirement not a issue.
> 
> Yes, but if you switch it to normal mode, why does the overview map on the user contribution list page get also switched to satellite mode? That's weird ;-)

Please guide me whats wrong in it.
check the gif here https://im7.ezgif.com/tmp/ezgif-7-104f32e8649d.gif

**Comment by norbertschuler** [comment]
- [ ] Just tested the latest version: The latest bugfixes introduced a new error not showing anymore the tree icon for my first user contribution in the maximized map or in the user contribution detail page:

Simulator Screen Shot - iPhone 11 - 2020-04-17 at 21 15 15

**Comment by haideralishah** [comment]
> * [ ]  I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:
> 
> Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 14
> Simulator Screen Shot - iPhone 11 - 2020-04-17 at 18 58 05

this is because of CDN problem as @sagararyal told me couple of days back.

**Comment by haideralishah** [comment]
> * [x]  Back button is often unresponsive in the Maps/ Single Contribution Page, Improve the button performance.
> * [x]  Clicking on tree icon on map, often takes to wrong/different contribution. This has been fixed for navigation from cards, but not from icons.
> * [x]  Loading a single contribution fetches all projects related to contributions. instead it should fetch neighbors only When contribution cards are stacked as `A [B] C` B being the current view, only load projects related to A, B & C, (performance)
> * [x]  Remove Current Location Icon from the Map View
> * [x]  Change to a icon of a tree for map marker instead of image (performance)
> * [ ]  Use Apple Maps in iOS ( if it is more performant than google maps)
> * [x]  disable the marker press if detail not show
> * [x]  revert the `headerText` on outer card list
> * [x]  disable arrows keys on both ends
> * [x]  clicking on multiple card on outer list
> * [x]  Indexing issues
> * [x]  In Single Contribution, enable Satellite View by default.
> 
> Things to consider:
> 
> * [x]  Look into Map Clustering
> 
> Fixes #1958
> 
> Changes in this pull request:
> 
> * Show road map instead of satellite
> * Center Map so that more than one items are visible.
> 
> @sagararyal

single tree icon is replaced with vector icon and double trees icons are still images. Because they are not available in vector icons.

apple map logo was causing problem, it's position does not change that is why not used.

clustering package or libraries have some limitations and can not be used with functionalities that are applied in map.

**Comment by norbertschuler** [comment]
> > * [ ]   I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:
> 
> this is because of CDN problem as @sagararyal told me couple of days back.

But how is the first screen editing the user contribution able to display the image if you can not?

**Comment by norbertschuler** [comment]
> * [x]  One more issue: When I use the button to maximize the map, I only get show the detailed map for my newest user contribution - is that's how it's wanted to be?

@haideralishah or @sagararyal can you answer the question why we only show the first user contribution when maximizing the overview map?

**Comment by norbertschuler** [comment]
> > > * [ ]    I uploaded an image to one of my registered trees, but it is not shown in the user contribution details (instead a white empty space is shown) - see the following two images:
> > 
> > 
> > this is because of CDN problem as @sagararyal told me couple of days back.
> 
> But how is the first screen editing the user contribution able to display the image if you can not?

Fix this myself.

**Comment by norbertschuler** [comment]
- [ ] On my Android emulator with a smaller screen I (1) was not able to scroll down the whole image for a multiple-tree registration and (2) also was not able to scroll down until the second measurement data of a single-tree registration:

Screenshot_1587211172

Screenshot_1587211207

**Comment by norbertschuler** [comment]
- [ ] I got this error on my Android emulator looking at a user contribution:

Screenshot_1587295714

**Comment by sanjayradadiya** [comment]
@haideralishah can you please check this  

- [ ]  map zoom out not working properly

>https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1995#issuecomment-601186037
>https://www.loom.com/share/e5135bf8df1c44eab8f1c4896fcb7303
image

- [ ] after delete tree when pressing go back button throwing scrollToIndex out of range
https://www.loom.com/share/7669ec320d164bffb6068656e1888f5f

- [ ] Icon not in center 
image

- [ ] when click on specific contribution map are not moveable is this is expected behaviour ?
https://www.loom.com/share/7f5f58f70c9f4862bbed82368243f951

**Comment by norbertschuler** [comment]
> * [x]  I got this error on my Android emulator looking at a user contribution:

Fixed this myself by adding the missing `getItemLayout` property to `FlatList` in order to enable proper `scrollToIndex` methods.

**Comment by norbertschuler** [comment]
> * [x]  I found these two warnings in the browser console when I first open my "Me" screen:
> 
> ```
> Warning: Failed child context type: Invalid child context `virtualizedCell.cellKey` of type `number` supplied to `CellRenderer`, expected `string`.
>     in CellRenderer (at VirtualizedList.js:767)
>     in RCTScrollContentView (at ScrollView.js:1038)
>     in RCTScrollView (at ScrollView.js:1178)
>     in ScrollView (at VirtualizedList.js:1183)
>     in VirtualizedList (at FlatList.js:676)
>     in FlatList (at UserHome/index.native.js:617)
> ```

Fixed this myself.

**Comment by norbertschuler** [comment]
- [ ] While I tried to reproduce the issues of @sanjayradadiya I also found another issue when you register your first tree:
```
TypeError: Cannot read property 'length' of null

This error is located at:
    in AnimatedViews (created by ConnectFunction)
    in ConnectFunction (at UserHome/index.native.js:278)
    in RCTView (at UserHome/index.native.js:269)
    in RCTView (at UserHome/index.native.js:526)
    in RCTScrollContentView (at ScrollView.js:1038)
    in RCTScrollView (at ScrollView.js:1151)
    in ScrollView (at UserHome/index.native.js:323)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (at react-native-safe-area-view/index.js:163)
    in SafeView (at withOrientation.js:54)
    in withOrientation (at UserHome/index.native.js:321)
    in UserHome (created by Context.Consumer)
    in withNavigation(UserHome) (at UserHome/index.js:44)
    in UserHomeContainer (created by ConnectFunction)
    in ConnectFunction (at AppDrawerNavigator.js:213)
    in Unknown (created by SceneView)
    in SceneView (created by TabNavigationView)
    in RCTView (created by ResourceSavingScene)
    in RCTView (created by ResourceSavingScene)
    in ResourceSavingScene (created by TabNavigationView)
    in RCTView (at screens.native.js:132)
    in ScreenContainer (created by TabNavigationView)
    in RCTView (created by TabNavigationView)
    in TabNavigationView (created by NavigationView)
    in NavigationView (created by Navigator)
    in Navigator (created by SceneView)
    in SceneView (created by CardContainer)
    in RCTView (created by CardContainer)
    in RCTView (created by CardContainer)
    in RCTView (created by Card)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (created by PanGestureHandler)
    in PanGestureHandler (created by Card)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (created by Card)
    in RCTView (created by Card)
    in Card (created by CardContainer)
    in CardContainer (created by CardStack)
    in RCTView (created by MaybeScreen)
    in MaybeScreen (created by CardStack)
    in RCTView (created by MaybeScreenContainer)
    in MaybeScreenContainer (created by CardStack)
    in CardStack (created by KeyboardManager)
    in KeyboardManager (created by Context.Consumer)
    in SafeAreaProviderCompat (created by StackView)
    in RCTView (created by StackView)
    in StackView (created by StackView)
    in StackView
    in Unknown (created by Navigator)
    in Navigator (created by SceneView)
    in SceneView (created by Drawer)
    in RCTView (created by ResourceSavingScene)
    in RCTView (created by ResourceSavingScene)
```
    in ResourceSavingScene (created by Drawer)
    in RCTView (at screens.native.js:132)
    in ScreenContainer (created by Drawer)
    in RCTView (at createAnimatedComponent.js:233)
    in AnimatedComponent(Component) (created by Drawer)
    in RCTView (at createAnimatedComponent.js:233)
    in AnimatedComponent(Component) (created by PanGestureHandler)
    in PanGestureHandler (created by Drawer)
    in Drawer (created by DrawerView)
    in DrawerView (created by Navigator)
    in Navigator (created by NavigationContainer)
    in NavigationContainer (at AppDrawerNavigatorContainer.js:102)
    in RCTView (at AppDrawerNavigatorContainer.js:101)
    in AppDrawerNavigatorContainer (created by ConnectFunction)
    in ConnectFunction (at TreeCounter.native.js:32)
    in RCTView (at createAnimatedComponent.js:151)
    in AnimatedComponent (at src/index.tsx:85)
    in SafeAreaView (at TreeCounter.native.js:28)
    in RNCSafeAreaView (at src/index.tsx:28)
    in SafeAreaProvider (at TreeCounter.native.js:27)
    in App (created by ConnectFunction)
    in ConnectFunction (at App/index.native.js:22)
    in GlobalErrorBoundary (at App/index.native.js:21)
    in Provider (at App/index.native.js:20)
    in App (at renderApplication.js:40)
    in RCTView (at AppContainer.js:101)
    in ChildrenWrapper (at wrapRootComponent.js:9)
    in _default (at wrapRootComponent.js:8)
    in Root (at AppContainer.js:115)
    in RCTView (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:39)

**Comment by norbertschuler** [comment]
- [ ] If you have two registered trees and delete one tree and go back to the overview you can get an exception:

```
Invariant Violation: scrollToIndex out of range: requested index 1 but maximum is 0

VirtualizedList.scrollToIndex
    index.bundle?platform=ios&dev=true&minify=false:58010:9
FlatList.scrollToIndex
    index.bundle?platform=ios&dev=true&minify=false:57636:25
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:285300:32
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:30719:21
_callTimer
    index.bundle?platform=ios&dev=true&minify=false:30636:9
Object.callTimers
    index.bundle?platform=ios&dev=true&minify=false:30843:9
MessageQueue.__callFunction
    index.bundle?platform=ios&dev=true&minify=false:2689:44
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:2402:17
MessageQueue.__guard
    index.bundle?platform=ios&dev=true&minify=false:2643:13

```

**Comment by norbertschuler** [comment]
- [ ] Also they is a crash when you try to open and select a project for a TPO while registering trees: [register.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4508170/register.mp4.zip)
```
TypeError: Cannot read property 'type' of undefined

<unknown>
    index.bundle?platform=ios&dev=true&minify=false:209228:27
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:209256:18
Dropdown.onSelect
    index.bundle?platform=ios&dev=true&minify=false:219002:11
DropdownItem.onPress
    index.bundle?platform=ios&dev=true&minify=false:221617:11
Button.onPress
    index.bundle?platform=ios&dev=true&minify=false:221758:11
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:221975:22
_callTimer
    index.bundle?platform=ios&dev=true&minify=false:30638:9
Object.callTimers
    index.bundle?platform=ios&dev=true&minify=false:30843:9
MessageQueue.__callFunction
    index.bundle?platform=ios&dev=true&minify=false:2689:44
<unknown>
    index.bundle?platform=ios&dev=true&minify=false:2402:17
```

**Comment by norbertschuler** [comment]
> * [x]  If you have two registered trees and delete one tree and go back to the overview you can get an exception:
> 
> ```
> Invariant Violation: scrollToIndex out of range: requested index 1 but maximum is 0

Hopefully fixed this.

**Comment by norbertschuler** [comment]
I extracted one issue as https://github.com/Plant-for-the-Planet-org/treecounter-app/issues/2218 as it also happens in the develop branch.

**Comment by norbertschuler** [comment]
> * [x]  While I tried to reproduce the issues of @sanjayradadiya I also found another issue when you register your first tree:  [register_first_tree.mp4.zip](https://github.com/Plant-for-the-Planet-org/treecounter-app/files/4508162/register_first_tree.mp4.zip)
> 
> ```
> TypeError: Cannot read property 'length' of null

I cannot reproduce this bug anymore after fixing https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1995#issuecomment-617203533
